### PR TITLE
chore: remove atomic<> from ReplicaInfo::state

### PR DESCRIPTION
**Pull request by romange**
This field is protected by ReplicaInfo::mu so non-protected access to it shows a design problem. Indeed, it was done for being able to access this field without a mutex inside ReplicationLags() function.

I moved the access to this field to GetReplicasRoleInfo where we need to lock ReplicaRoleInfo anyways. Also, done some cleanups in the file.

Finally, raised a threshold for "tx queue too long" warnings.

### CODE REVIEW

File: src/server/dflycmd.cc [hunk 1]
```diff
@@ -764,18 +711,15 @@ std::map<uint32_t, LSN> DflyCmd::ReplicationLags() const {
   shard_set->RunBriefInParallel([&shard_lags, this](EngineShard* shard) {
     auto& lags = shard_lags[shard->shard_id()];
     for (const auto& info : replica_infos_) {
-      if (info.second->state.load() != SyncState::STABLE_SYNC) {
-        lags[info.first] = std::numeric_limits<LSN>::max();
-        continue;
+      const ReplicaInfo* replica = info.second.get();
+      if (shard->journal()) {
+        int64_t lag = shard->journal()->GetLsn() - replica->flows[shard->shard_id()].last_acked_lsn;
+        lags[info.first] = lag;
```
Comment (dranikpg) [inline src/server/dflycmd.cc:717]: I'd suggest adding a `DCHECK(!replica.mu.try_lock())` to verify it's locked when the code around changes

File: src/server/dflycmd.cc [hunk 2]
```diff
@@ -700,14 +636,22 @@ shared_ptr<DflyCmd::ReplicaInfo> DflyCmd::GetReplicaInfo(uint32_t sync_id) {
...
+  lock_guard lk(mu_);
 
+  vec.reserve(replica_infos_.size());
   auto replication_lags = ReplicationLags();
 
   for (const auto& [id, info] : replica_infos_) {
+    LSN lag = replication_lags[id];
+
+    lock_guard repl_lk{info->mu};
+
```
Comment (dranikpg) [inline src/server/dflycmd.cc:648]: the problem here is that the full sync -> stable sync transition might take some time (we should actually measure it), so even if only one replica takes unusually longer, your stats will freeze for _all_ replicas... Given we do it in GetMetrics(), INFO ALL will freeze as well

File: src/server/dflycmd.cc (hunk 1 above)
Comment (romange) [reply src/server/dflycmd.cc:717]: it's not locked though and we do not access any of the fields that can be accessed by multiple threads.

Comment (dranikpg) [reply src/server/dflycmd.cc:723]: Ok, agree. ReplicaInfo::flows is pre-sized and last_acked_lsn is only touched during stable sync from it's respective thread

File: src/server/dflycmd.cc [hunk 3]
```diff
@@ -700,14 +636,28 @@ shared_ptr<DflyCmd::ReplicaInfo> DflyCmd::GetReplicaInfo(uint32_t sync_id) {
...
+      state = info->replica_state;
+      // If the replica is not in stable sync, its lag is undefined, so we set it as max.
+      if (state != SyncState::STABLE_SYNC) {
+        lag = std::numeric_limits<LSN>::max();
+      }
+      info->mu.unlock();
+    } else {
+      lag = std::numeric_limits<LSN>::max();
+    }
+    vec.push_back(ReplicaRoleInfo{info->address, info->listening_port, SyncStateName(state), lag});
```
Comment (dranikpg) [inline src/server/dflycmd.cc:660]: It's safe but possibly some monitoring setups can get confused by this preparation state 🙂

**Review by dranikpg — APPROVED** [review]

**Pull request merged**